File: spring-cli-app-java/src/main/java/com/steeplesoft/spring_cli_app_java/Example2Command.java
package com.steeplesoft.spring_cli_app_java;

import com.beust.jcommander.Parameters;
import org.springframework.stereotype.Component;

@Component
@Parameters(commandNames = Example2Command.COMMAND_NAME,
        commandDescription = "Example command #2")
public class Example2Command implements Command {
    public static final String COMMAND_NAME = "something-else";

    @Override
    public String commandName() {
        return COMMAND_NAME;
    }

    @Override
    public void run() {
        System.out.println("You ran something else!");
    }
}

File: spring-cli-app-kotlin/src/main/kotlin/com/steeplesoft/spring_cli_app/Example2Command.kt
package com.steeplesoft.spring_cli_app

import com.beust.jcommander.Parameters
import org.springframework.stereotype.Component

@Component
@Parameters(commandNames = [Example2Command.COMMAND_NAME], commandDescription = "Example command #2")
class Example2Command : Command {
    override fun commandName(): String {
        return COMMAND_NAME
    }

    override fun run() {
        println("You ran something else!")
    }

    companion object {
        const val COMMAND_NAME = "something-else"
    }
}
